integration-tests/src/tests/client/resharding.rs

```diff
@@ -1477,6 +1488,141 @@ fn test_shard_layout_upgrade_cross_contract_calls_v3_seed_44() {
     test_shard_layout_upgrade_cross_contract_calls_impl(ReshardingType::V3, 44);
 }
 
+fn setup_test_env_with_promise_yield_txs(
+    test_env: &mut TestReshardingEnv,
+    epoch_length: u64,
+) -> Vec<CryptoHash> {
+    let genesis_hash = *test_env.env.clients[0].chain.genesis_block().hash();
+
+    // Generates an account for each shard
+    let contract_accounts = gen_shard_accounts();
+
+    // Add transactions deploying nightly_rs_contract to each account
+    let mut init_txs = vec![];
+    for account_id in &contract_accounts {
+        let signer =
+            InMemorySigner::from_seed(account_id.clone(), KeyType::ED25519, account_id.as_ref());
+        let actions = vec![Action::DeployContract(DeployContractAction {
+            code: near_test_contracts::nightly_rs_contract().to_vec(),
+        })];
+        let init_tx = SignedTransaction::from_actions(
+            1,
+            account_id.clone(),
+            account_id.clone(),
+            &signer,
+            actions,
+            genesis_hash,
+        );
+        init_txs.push(init_tx);
+    }
+    test_env.set_init_tx(init_txs);
+
+    let mut yield_tx_hashes = vec![];
+    let mut nonce = 100;
+
+    // In these tests we set the epoch length equal to the yield timeout length.
+    assert!(
+        epoch_length
+            == RuntimeConfig::test().wasm_config.limit_config.yield_timeout_length_in_blocks,
+    );
+    // Add transactions invoking promise_yield_create near the epoch boundaries.
+    for height in [
+        epoch_length - 2, // create in first epoch, trigger timeout during resharding epoch
+        epoch_length - 1,
+        epoch_length,
+        2 * epoch_length - 2, // create during resharding epoch, trigger timeout on upgraded layout
+        2 * epoch_length - 1,
+        2 * epoch_length,
+        3 * epoch_length - 2, // both the create and the timeout will occur on upgraded layout
+        3 * epoch_length - 1,
+        3 * epoch_length,
+    ] {
+        let txs = contract_accounts
```

Comment:
small nit: this would be more readable as a for loop and in a dedicated helper method